src/pages/Governance/components/snackbar/TransactionResponseSnackbar.tsx

```diff
@@ -0,0 +1,83 @@
+import React from "react";
+import {Types} from "aptos";
+import {IconButton} from "@mui/material";
+import CloseIcon from "@mui/icons-material/Close";
+import {
+  TransactionResponse,
+  TransactionResponseOnError,
+  TransactionResponseOnSubmission,
+} from "../../../../api/hooks/useSubmitTransaction";
+import {useGetTransaction} from "../../../../api/hooks/useGetTransaction";
+import SuccessSnackbar from "./SuccessSnackbar";
+import FailureSnackbar from "./FailureSnackbar";
+import ErrorSnackbar from "./ErrorSnackbar";
+
+export function CloseAction({onCloseSnackbar}: {onCloseSnackbar: () => void}) {
+  return (
+    <IconButton
+      size="small"
+      aria-label="close"
+      color="inherit"
+      onClick={onCloseSnackbar}
+    >
+      <CloseIcon fontSize="small" />
+    </IconButton>
+  );
+}
+
+type TransactionResponseSnackbarProps = {
+  transactionResponse: TransactionResponse | null;
+  onCloseSnackbar: () => void;
+  refreshOnSuccess?: boolean;
+};
+
+export default function TransactionResponseSnackbar({
+  transactionResponse,
+  onCloseSnackbar,
+  refreshOnSuccess,
+}: TransactionResponseSnackbarProps) {
+  if (transactionResponse == null) {
+    return null;
+  }
+
+  if (!transactionResponse.transactionSubmitted) {
+    return (
+      <ErrorSnackbar
+        errorMessage={
+          (transactionResponse as TransactionResponseOnError).message
```

Comment:
Remove the type cast. Typescript should be able to infer that it's `TransactionResponseOnError` based on the `if (!transactionResponse.transactionSubmitted)` conditional.

Reply:
Oh nice! Didn't know that.